default/scripting/ship_parts/FighterHangar/FT_HANGAR_3.focs.txt

```diff
@@ -5,6 +5,10 @@ Part
     class = FighterHangar
     capacity = 2
     damage = 5
+    combatTargets = And [
+        [[COMBAT_TARGETS_VISIBLE_ENEMY]]
+        Or [ Ship Fighter ]
```

Comment:
I think bombers (and heavy bombers) should not target Fighters.
We could add another layer which targets Fighters if there are no Ships if you prefer, but i do not think this is really necessary.
There is an efficient but unprecise FOCS implementation for the target-fighters-iff-no-ships condition:
```
And [
    [[COMBAT_TARGETS_VISIBLE_ENEMY]]
    Or·[
        Ship
        // probability to hit fighter is only high if there are few enemy ships
        NumberOf·number·=·1·condition·=·Fighter
    ]
]
```
Or a precise, more verbose and maybe costly(?) variant which needs to calculate the number of enemy ships:
```
And [
    [[COMBAT_TARGETS_VISIBLE_ENEMY]]
    Or [
        Ship
        And [
            NumberOf·number·=·1·condition·=·Fighter
            Number low = 0 high = 0 condition = And [
                Object id = RootCandidate.ID
                [[COMBAT_TARGETS_VISIBLE_ENEMY]] // not sure if this needs to be reevaluated
                Ship
            ]
         ]
     ]
]
```

Reply:
such tweaks / balance / optimization to parts can be done after the mechanisms to allow them are in place

Reply:
>>    Do you intend to specialise the fighter combatTargets in this branch?
>
> Hadn't really thought about it. Feel free to suggest stuff.

Working on the parts at least checks if the mechanism can fullfill the intentions and if it actually works